sdk/include/reactos/debug.h

```diff
@@ -58,6 +61,74 @@ RtlAssert(
     _In_opt_z_ PCHAR Message
 );
 
+#elif defined(_WIN32)
+/* Win32 User-Mode */
+
+#ifndef DEBUG_BUFSIZE
+#define DEBUG_BUFSIZE 512
+#endif
+
+static inline
+ULONG
+__cdecl
+W32DbgPrint(_In_z_ _Printf_format_string_ PCSTR Format, ...)
+{
+    va_list va;
+    char szBuff[DEBUG_BUFSIZE];
+
+    va_start(va, Format);
+
+#ifdef _STRSAFE_H_INCLUDED_
+    StringCchVPrintfA(szBuff, _countof(szBuff), Format, va);
+#else
+    _vsnprintf(szBuff, _countof(szBuff), Format, va);
+#endif
+
+    OutputDebugStringA(szBuff);
+    va_end(va);
+
+    return 0; /* Return STATUS_SUCCESS, since we are supposed to mimic DbgPrint */
+}
+
+#define DbgPrint(Format, ...) W32DbgPrint(Format, __VA_ARGS__)
+#define DbgPrintEx(ComponentId, Level, Format, ...) W32DbgPrint(Format, __VA_ARGS__)
+
+static inline
+__analysis_noreturn
+VOID
+NTAPI
+W32Assert(
+    _In_ PVOID FailedAssertion,
+    _In_ PVOID FileName,
+    _In_ ULONG LineNumber,
+    _In_opt_z_ PCHAR Message)
+{
+    INT id;
+    char szBuff[DEBUG_BUFSIZE];
+
+#ifdef _STRSAFE_H_INCLUDED_
+    StringCchPrintfA(szBuff, _countof(szBuff),
+#else
+    _snprintf(szBuff, _countof(szBuff),
+#endif
+              "*** Assertion failed: %s%s\n"
+              "***   Source File: %s, line %lu",
+              (Message ? Message : ""), (PSTR)FailedAssertion, (PSTR)FileName, LineNumber);
+
+    id = MessageBoxA(NULL, szBuff, "Assertion Failure", MB_ICONERROR | MB_ABORTRETRYIGNORE);
```

Comment:
For the messagebox flags I would suggest: `MB_ABORTRETRYIGNORE | MB_ICONERROR | MB_SETFOREGROUND | MB_TASKMODAL` (like in the CRT debugging).
The 3rd flag puts the msgbox in front when it appears, and the last flag makes that "all the top-level windows belonging to the current thread [of the current application] are disabled if the hWnd parameter is NULL", and since the app is crashing, it should "prevent input to other windows in the calling thread without suspending other threads."
(I'm quoting MSDN here.)